castles.ts

```diff
@@ -0,0 +1,245 @@
+//% color="#AA278D" icon="\uf447"
+namespace castles {
+    /**
+     * Build a castle wall of specified width, length, and height out of blockType
+     * @param width how thick the wall will be. If at least 3, there will be a gap for inner and outer walls
+     * @param length how far a wall will span in blocks
+     * @param height the number of blocks high the wall will stand
+     * @param blockType the Minecraft block the wall will be built with
+     */
+    //% blockId=castles_build_wall
+    //% block="build castle wall|made of $blockType width $width length $length height $height" 
+    //% width.defl=3
+    //% width.min=2 width.max=1000
+    //% length.defl=27
+    //% length.min=4 length.max=1000
+    //% height.defl=6
+    //% height.min=1 height.max=1000
+    //% blockType.shadow=minecraftBlock
+    //% blockType.defl=Block.Cobblestone
+    //% weight=340
+    //% help=github:makecode-minecraft-castle-builder/docs/castle-wall
+    export function buildCastleWall(width: number, length: number, height: number, blockType: Block) {
```

Comment:
the type of `blockType` should be number, otherwise BlocklyLoader will try to initialize it as a dropdown and ignore the shadow. same goes for all of the other functions